src/App.tsx

```diff
@@ -1,26 +1,289 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  ChangeEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 
-const USER_ID = 0;
+import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  uploadTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { ErrorType } from './types/Errors';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { emptyTodo } from './utils/EmptyTodo';
+import { UpdateTodoData } from './types/UpdateTodoData';
+import { ErrorNotification } from './components/ErrorNoti/ErrorNotiflication';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodoStatus, setSelectedTodoStatus] = useState(TodoStatus.All);
+  const [errorMessage, setErrorMessage] = useState<ErrorType | ''>('');
+  const [processingTodos, setProcessingTodos] = useState<number[]>([]);
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const focusInputField = () => inputRef.current?.focus();
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage(ErrorType.LOAD_TODOS));
+  }, []);
+
+  useEffect(() => {
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.